### DISCUSSION TITLE
Is docker image works on mac m1?

### DISCUSSION THREAD

**Discussion by skorphil** [v2fly/v2ray-core#2859] [Q&A]
I have tried to run the container https://hub.docker.com/r/v2fly/v2fly-core
Exits immediately and does not start

**Comment by skorphil** [comment]
Well, that is not due to M1, that's just behavior of container https://github.com/v2fly/v2ray-core/discussions/2860

**Comment by fanganpai** [comment]
docker的原理是在本机的linux内核上通过重新挂载root、改变namespace和cgroup实现的隔离，本质上程序仍然需要跑在你自己的Linux内核上，因此不可能在不具有Linux内核的操作系统上运行。windows上docker的实际上也是运行在wsl中的。如果你能在mac上提供一个linux虚拟机的话或许可以
